### PR TITLE
LPCXpresso55S69 board

### PR DESCRIPTION
**Pull request by agansari**
Adding LPC55S69 port to Zephyr RTOS.
 - mcu initialization
 - uart support
 - gpio pins and leds
 - board documentation
 - how to setup jlink and daplink(flash copy only)

Fixes #15647

### CODE REVIEW

File: boards/arm/lpcxpresso55S69/board.cmake
Comment (galak) [inline boards/arm/lpcxpresso55S69/board.cmake]: I found that 'LPC55S69_core0' and 'LPC55S69_core1' I think work for jlink.

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: Section heading underline needs to be at least as long as the title (add some more ``---``)

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: add a colon at the end:  ``.. _updating LPCXpresso firmware:``

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: If you need to have an underscore here, write this as ``PIO1_7\_`` (same on line 88 above).  Otherwise delete both of these trailing underscores.

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: Need to also add a link for ``updating LPCXpresso firmware`` you're referencing on line 137 above.

File: doc/guides/debugging/probes.rst
Comment (dbkinder) [inline doc/guides/debugging/probes.rst]: This would render better as:
```
.. note:: Verify the firmware supports your board by visiting `Firmware for LPCXpresso`_
```

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: rename the ``lpc55s69_evk.png`` image included with this PR to ``lpcxpresso55s69.png`` (our convention is to have the board, board doc, and board image name be the same).

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]:
```suggestion
subsystem, and accelerometer, with several options for adding off-the-shelf add-on
```

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]:
```suggestion
boards for networking, sensors, displays, and other interfaces.
```

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]:
```suggestion
For more information about the LPC55S69 SoC and LPCXPRESSO55S69 board, see:
```

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: Change to:
```
The default configuration file
``boards/arm/lpcxpresso55s69/lpcxpresso55s69_cpu0_defconfig``
only enables the first core.
```

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]:
```suggestion
PyOCD support for this board is ongoing and not yet available.
```

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (dbkinder) [inline boards/arm/lpcxpresso55S69/doc/index.rst]: Let's merge these two sentences into one "step":
```
Connect a USB cable from your PC to P6, and use the
serial terminal of your choice (minicom, putty, etc.)
with the following settings:
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Missed clicking on Request changes :)

File: boards/arm/lpcxpresso55S69/doc/index.rst
Comment (agansari) [reply boards/arm/lpcxpresso55S69/doc/index.rst]: No need, the default is now JLink. DapLink is not yet fully working, the tutorial is upgrading the debug firmware is there to let you know you have other options, but tools are under development.

I added "This example uses the :ref:`jlink-debug-host-tools` as default. to both flashing and debugging". What do you think?

File: soc/arm/nxp_lpc/lpc55xxx/Kconfig.series
Comment (ioannisg) [inline soc/arm/nxp_lpc/lpc55xxx/Kconfig.series]: If the series only include CM33 variants, CPU_HAS_SYSTICK is not needed. Actually the option is now called CPU_CORTEX_M_HAS_SYSTICK

File: soc/arm/nxp_lpc/lpc55xxx/Kconfig.soc
Comment (ioannisg) [inline soc/arm/nxp_lpc/lpc55xxx/Kconfig.soc]: Could this selection be moved in the _SERIES? It looks common to the tw CPUs.

**Review by ioannisg — CHANGES_REQUESTED** [review]
Looks ok, I left a couple of comments.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by agansari** [comment]
Thank you guys @MaureenHelm @galak @ioannisg @dbkinder made the changes you required. Can you have another look?

File: soc/arm/nxp_lpc/Kconfig
Comment (MaureenHelm) [inline soc/arm/nxp_lpc/Kconfig]: Don't remove existing copyrights. I meant in https://github.com/zephyrproject-rtos/zephyr/pull/15647#discussion_r279887544 to update the copyright in new files.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please make the `fsl_{power,reset}` changes for lpc54114 and update `ext/hal/nxp/mcux/scripts/import_mcux_sdk.py` in a separate commit before you import the sdk for lpc55s69.

File: soc/arm/nxp_lpc/Kconfig
Comment (agansari) [reply soc/arm/nxp_lpc/Kconfig]: @MaureenHelm made the changes you required

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Looks much better, thank you. Please start working to resolve the CI failures, except for the license check which is expected to fail due to new files being imported into `ext/`.

File: ext/hal/nxp/mcux/devices/LPC55S69/fsl_power.c
Comment (agansari) [reply ext/hal/nxp/mcux/devices/LPC55S69/fsl_power.c]: Yes, I see this now, didn't pay attention when resetting the files. I used git mv, but as move is in one commit and delete in another, the files were duplicated. Need to do another forcepush besides the one below.

Also I think I found the issue with failed CI, it was related to CMakeLists.txt, couldn't build other mcux devices as fsl_reset and fsl_power were expected in device folder for Kinetis+i.MX as well.

Comment (agansari) [reply ext/hal/nxp/mcux/devices/LPC55S69/fsl_power.c]: @MaureenHelm fixed all the bad file versioning issues in my former commit and automated checks now pass, except for licence.

Comment (ioannisg) [reply ext/hal/nxp/mcux/devices/LPC55S69/fsl_power.c]: This looks merge-able now, IMHO

**Review by ioannisg — APPROVED** [review]

File: ext/hal/nxp/mcux/devices/LPC55S69/fsl_powerquad_basic.c
Comment (MaureenHelm) [inline ext/hal/nxp/mcux/devices/LPC55S69/fsl_powerquad_basic.c]: powerquad should go in the shared drivers folder instead

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
@dbkinder please revisit

File: ext/hal/nxp/mcux/devices/LPC55S69/fsl_powerquad_basic.c
Comment (agansari) [reply ext/hal/nxp/mcux/devices/LPC55S69/fsl_powerquad_basic.c]: Thank you @MaureenHelm powerquad was filtered as part of fsl_power by the script. Waiting for a review from @dbkinder and @MeganHansen for another force push.

**Review by galak — DISMISSED** [review]
some minor cleanups.

File: dts/arm/nxp/nxp_lpc55S6x.dtsi
Comment (galak) [inline dts/arm/nxp/nxp_lpc55S6x.dtsi]: we need to look at structuring this similar to how dts/arm/nordic/nrf9160* is done.  So that secure/non-secure can work cleanly in the future.

Comment (agansari) [reply dts/arm/nxp/nxp_lpc55S6x.dtsi]: @galak i would make a separate issue from this pull request to separate secure and non-secure configuration. I also think that it can be done different from nrf9160 because sec/nonsec HW addresses for memory and peripherals are offseted by 0x10000000.

Comment (galak) [reply dts/arm/nxp/nxp_lpc55S6x.dtsi]: Ok, can we at least use the same address range for all the peripherals?

File: dts/arm/nxp/nxp_lpc55S6x.dtsi
Comment (galak) [inline dts/arm/nxp/nxp_lpc55S6x.dtsi]: Can we make this `0x5008c000` instead?

File: dts/arm/nxp/nxp_lpc55S6x.dtsi
Comment (galak) [inline dts/arm/nxp/nxp_lpc55S6x.dtsi]: Can we make this `0x5008c000`

Comment (agansari) [reply dts/arm/nxp/nxp_lpc55S6x.dtsi]: @galak moved all memory and peripherals to secure address space, validated and force pushed.

Also created #16330 to have secure/non-secure split.

**Review by dbkinder — APPROVED** [review]
doc change LGTM, thanks.

File: ext/hal/nxp/mcux/README
Comment (MaureenHelm) [inline ext/hal/nxp/mcux/README]: Per https://github.com/zephyrproject-rtos/zephyr/pull/16049#discussion_r283954331, please use the "revision" attribute instead of the tag.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Just one nit, otherwise LGTM

File: ext/hal/nxp/mcux/README
Comment (agansari) [reply ext/hal/nxp/mcux/README]: @MaureenHelm done!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
